cloudify_agent/operations.py

```diff
@@ -48,6 +49,11 @@
 
 @operation
 def install_plugins(plugins, **_):
+    tenant_mask = False
+    if ctx.tenant_name != workflow_ctx.tenant_name:
```

Comment:
The problem is that any updates to execution status (e.g. if the workflow fails) will still require this same logic. See comments on https://github.com/cloudify-cosmo/cloudify-plugins-common/pull/338 for why I think we probably don't want to bed this in much further than it is.